### PR TITLE
[chore][Testbed] - Try flushing the data before exiting in load generator

### PR DESCRIPTION
**Pull request by VihasMakwana**
**Description:** In the `LoadGenerator`, if we encounter a non-permanent error and are in the process of retrying, we should not immediately exit upon receiving a stop signal. Instead, we need to first flush the existing data and then proceed to exit the LoadGenerator. This is necessary for stress test cases and we need to validate the data received and data sent.

### CODE REVIEW

File: testbed/testbed/load_generator.go
Comment (bacherfl) [inline testbed/testbed/load_generator.go:261]: would it be possible to change this to still exit this function eventually after a certain number of retries? Otherwise, some cases like [the one testing the MemoryLimiter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c1e2df6b5a5b7397a168c8b46f72db26ec3dd3af/testbed/tests/trace_test.go#L196) will stall infinitely - In the linked example the collector at one point rejects any incoming data items due to the memory limit being exceeded (which is a non permanent error) and the test does not terminate

Comment (VihasMakwana) [reply testbed/testbed/load_generator.go:261]: That's a great suggestion.
I'll implement a timeout based mechanism. Thanks for your review!

Comment (bacherfl) [reply testbed/testbed/load_generator.go:261]: thanks for the adaptation! just tried it out again with the test case I mentioned earlier and it seems to work well

Comment (VihasMakwana) [reply testbed/testbed/load_generator.go:261]: Glad to know it worked. Thanks for your quick review!

File: testbed/testbed/test_bed.go
Comment (bacherfl) [inline testbed/testbed/test_bed.go:63]: i guess this can be reverted

File: testbed/tests/trace_test.go
Comment (bacherfl) [inline testbed/tests/trace_test.go:194]: this can be reverted as well

File: testbed/testbed/test_bed.go
Comment (VihasMakwana) [reply testbed/testbed/test_bed.go:63]: yeah I pushed this by mistake

File: testbed/tests/trace_test.go
Comment (VihasMakwana) [reply testbed/tests/trace_test.go:194]: same as above

**Review by bacherfl — APPROVED** [review]

**Comment by bacherfl** [comment]
I think the title of the PR can be changed to start with `[chore]`, to skip the changelog check (which afaik is not required for test code)

**Comment by VihasMakwana** [comment]
@TylerHelmuth @open-telemetry/collector-contrib-approvers can some take a look at this?

**Comment by VihasMakwana** [comment]
@open-telemetry/collector-contrib-approvers can I get a quick approval here? this is needed for https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35209

**Review by djaglowski — APPROVED** [review]

**Pull request merged**
